#!/usr/bin/env python

PACKAGE = "opencv_apps"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_camera_info", bool_t, 0, "Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used.", False)

gen.add("r_limit_max", int_t, 0, "The maximum allowed field value Red", 255, 0, 255)
gen.add("r_limit_min", int_t, 0, "The minimum allowed field value Red", 0, 0, 255)
gen.add("g_limit_max", int_t, 0, "The maximum allowed field value Green", 255, 0, 255)
gen.add("g_limit_min", int_t, 0, "The minimum allowed field value Green", 0, 0, 255)
gen.add("b_limit_max", int_t, 0, "The maximum allowed field value Blue", 255, 0, 255)
gen.add("b_limit_min", int_t, 0, "The minimum allowed field value Blue", 0, 0, 255)

exit(gen.generate(PACKAGE, "color_filter", "RGBColorFilter"))
